[Target size]: UA controlled rule (#2182)
auto-wcag/auto-wcag: pages/glossary/can-be-targeted-by-pointer-event.md

---
title: Can be target by a pointer event
key: can-be-targeted-by-pointer-event
unambiguous: true
objective: true
input_aspects:
  - CSS styling
  - DOM tree
---

An element _can be targeted by pointer events_ when all the following conditions are true:

> - the element is a [semantic `widget`][semantic role]; and
> - the element is [focusable][]; and

Reviewer: I'm not sure both of these need to be true. At least one I would say.

```suggestion
- the element is a [semantic `widget`][semantic role] or is [focusable][]; and
```

> - the element [computed value][] of the `pointer-events` CSS property is `auto`; and

Reviewer 2: Do we need to consider 'stroke' and 'fill'? These are valid for SVG elements. See https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events

> - through scrolling, it is possible to have parts of the element's [border box][] which intersect the viewport and are not entirely covered by the [border boxes][] of elements with greater [computed][computed value] `z-index` and a [computed][computed value] `pointer-events` of `auto`.

Reviewer: This is tough to parse. How about:

```suggestion
- through scrolling, it is possible to have parts of the element's [border box][] appear within the viewport; and
- while scrolled within the viewport the element is not obscured by another element with a [computed][computed value] `pointer-events` of `auto`.
```

> 

Reviewer 3: Should we add a bullet point to clarify that the element must be visible (i.e., it is not set to display:none or visibility:hidden, it is not positioned out of the viewport such as left:-99999px, and its computed width and height are greater than 0)?

Without this clarification, it still makes sense theoretically, since the element can be targeted by a pointer event. However, if the element is not visible, this condition becomes practically useless as it cannot be targeted by a pointer device.

Or perhaps it is fine to keep it as is, but this statement should be defined within the applicability section for each rule.

#### Background

This definition tries to capture which HTML elements can actually react to pointer events. It is not possible to have an exact definition of these for two main reasons:

- Sometimes, the element that handles the event is not the element that appear to react to it, but an ancestor (or descendant) capturing the event during propagation or bubbling. In the most extreme case, the `body` element of a page could be the only one with an event handler, acting differently depending on where the event actually occurred. In such a case, a button would be perceived by users as something that can be targeted by a pointer event, while technically it is the `body` element which is targeted.

> - It is not possible to query the list of event listeners on a given elements. Some User Agents offer way to monitor events fired at a given element, but none offer a way to query for event listeners. Additionally, an event listener might ultimately do nothing and thus, for users, the corresponding element wouldn't look like it can be targeted by pointer events (since it effectively wouldn't react to them).

Reviewer: That's not quite true. Most browser devtools panels can show this. There just isn't a built in DOM method.

As a consequence, this definition has these two known limitations:

- Not all HTML elements that can actually be targeted by a pointer event match this definition. For example, an author may build custom buttons without giving them an appropriate role or making them [focusable][]; or content overflowing the [border box][] of an element is clickable, but that element might not match the definition if the [border box][] is off-screen. Elements that can actually be targeted by pointer events but do not match this definition likely fail either [Success Criterion 4.1.2 Name, Role, Value][sc412] or [Success Criterion 2.1.1 Keyboard][sc211].
- Not all HTML elements that match this definition can actually be targeted by a pointer event. For example, when the actual clickable area does not cover the full [border box][] and is entirely covered by other elements, or when the element has an event handler that does nothing. Elements that match this definition but cannot actually be targeted by pointer events likely fail [Success Criterion 2.5.6 Concurrent Input Mechanisms][sc256].

[border box]: https://www.w3.org/TR/css-box-3/#border-box 'CSS definition of Border Box'

> [computed value]: https://www.w3.org/TR/css-cascade-3/#computed 'CSS definition of Computed Value'

Reviewer 3: ```suggestion
[computed value]: https://www.w3.org/TR/css-cascade-5/#computed 'CSS definition of Computed Value'
```

updated with last definition
